### PR TITLE
update install python controller command to use api directly and remove controller loaded context

### LINKED ISSUE #11647: Change how context key `jupyter.controllersLoaded` is updated
@IanMatthewHuff 
As per discussion I think we can:
* Directly use the Python API to determine if python envs are available or not instead of checking the controllers.
* Remove the code to update `jupyter.controllersLoaded` from the `ControllerLoaded` class

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
Fixes #11647 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for feature-requests.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/notebooks/controllers/commands/installPythonControllerCommands.ts [hunk 1]
```diff
@@ -109,7 +107,7 @@ export class InstallPythonControllerCommands implements IExtensionSingleActivati
                     return;
                 }
 
-                if (!this.foundPythonConnections) {
+                if (this.interpreterService.environments.length === 0) {
                     // Extension is installed, but we didn't find any python connections
                     // recommend installing python in this case
                     await this.showInstallPythonExtensionContext.set(false);
```
Comment (DonJayamanne) [inline src/notebooks/controllers/commands/installPythonControllerCommands.ts:113]: I don't get code.
Shouldn't this be `this.interpreterService.environments.length > 0`

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
Need clarification on the logic

File: src/notebooks/controllers/commands/installPythonControllerCommands.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/notebooks/controllers/commands/installPythonControllerCommands.ts:113]: @DonJayamanne the code in this block should be hit if:

1. Python Extension is installed
2. We don't have any valid python environments to use (length of envs === 0)

I would think your suggestion would be saying if we did find environments to use. The translation from previous code is from 'did not find python connections' to 'did not find python environments'. Or am I misunderstanding?

Comment (DonJayamanne) [reply src/notebooks/controllers/commands/installPythonControllerCommands.ts:113]: Oh ok, yes its correct, the next line is `await this.showInstallPythonContext.set(true);`
& Github conviniently failed to display that crucial line.
All good.

**Review by DonJayamanne — APPROVED** [review]

File: src/notebooks/controllers/commands/installPythonControllerCommands.ts (hunk 1 above)
Comment (IanMatthewHuff) [reply src/notebooks/controllers/commands/installPythonControllerCommands.ts:113]: No problem! I just always worry that I flubbed the logic for things like this, so I was looking it over again and again 😆 .

**Pull request merged**